src/neptune/typing.py

```diff
@@ -80,3 +84,6 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
     @abc.abstractmethod
     def update(self, *, by: int, total: Optional[int] = None) -> None:
         ...
+
+
+ProgressBarType: TypeAlias = Optional[Union[bool, Type[ProgressBarCallback]]]
```

Comment:
I suggest unwrapping `ProgressBarType` from `Optional` as its kinda unusual practice. I think that

```
def f(x: Optional[ProgressBarType] = None)
    pass
```
is more clear then

```
def f(x: ProgressBarType = None)
    pass
```

Reply:
Good point. Done :)